_posts/2021-05-16-overview-of-architecture-review-processes.md

```diff
@@ -0,0 +1,253 @@
+---
+layout: post
+date:   2021-05-29 09:02:08 +0100
+title: "Overview of different Architecture Review Processes"
+categories:
+  - Architecture
+  - Organization
+  - Architecture Review Process
+published: true
+---
+Recently, I have been asked about what software architecture review processes exist in order to better steer architecture.
+An architecture review aims at different goals such as finding software design issues early in the development before they get costly.
+Architecture review processes, for example, formalize different steps.
+They, furthermore, define input and output to/from architecture reviews.
+This article describes the different architecture review process approaches.
+
+## What are Architecture Reviews and Architecture Review Processes?
+
+An architecture review, in a nutshell, is a mechanism for increasing the likelihood that a software/system architecture will be complete, consistent, and, thus, good {% cite Maranzano2005 %}.
+A good software architecture is important, otherwise the development can become slower.
+It can get more expensive to add new capabilities in the future.
+High internal quality, on the other side, can lead to faster delivery of new features; simply, because there is less cruft to get in the way {% cite Fowler2019 %}.
+
+Essentially, architecture reviews aim at {% cite Maranzano2005 %}:
+  1. finding software design problems early in development before they get costly.
+  1. building projects based on best practices and transfer this knowledge across the organization.
+  1. improving the organization's software quality and operations (-> documentation).
+
+Therefore, architecture review processes formalize different steps, the parties involved, or input and output to architecture reviews.
+Over the last decades, different architecture review processes have emerged.
+
+As every such process can be modified and adapted to the unique requirements of a specific company, we will focus on three different prototypical processes that are described in literature:
+  1. The "Classical" Architecture Review process explained by Maranzano et al. in {% cite Maranzano2005 %}.
+  1. The Architecture Decision Records initially described by Nygard in {% cite Nygard2011 %}.
+  1. The Lightweight Request for Comment (RFC)/Design Document approach explained in {% cite Ubl2020 %}, {% cite Winters2020 %}, {% cite Orosz2020 %}, {% cite Orosz2021 %}, {% cite Zimmermann2019 %}, {% cite Gonchar2020 %}, and {% cite Mozilla2020 %} from the perspectives of different organizations - sometimes called Lightweight RFC and sometimes Design Document approach. The approach has also been mentioned in the ThoughWorks Tech Radar Vol. 24 trial area as Lightweight RFC approach in {% cite ThoughtWorks2021 %}.
+
+In the next sections, we will describe these three different prototypical architecture review processes.
+
+## The "Classical" Architecture Review Process
```

Comment:
In comparison to the "Architecture Decision Records" chapter, this is kinda hard to read.
An info graphic will be 100% beneficial.